getdata_HN.py

```diff
@@ -70,29 +50,17 @@ def main():
         new_stories = {}
         with open(DATAFILE) as json_data:
             old_stories = json.load(json_data)
-            old_ids = old_stories.keys()
-            new_stories = old_stories
-            if _get_stories_id():
-                for new_story_id in _get_stories_id():
-                    for old_story_id in list(old_ids):
-                        if new_story_id != old_story_id:
-                            parameter = 'item/{}.json'\
-                                        '?print=pretty'.format(new_story_id)
-                            new_story = _get_json(parameter)
-                            if 'url' in new_story:
-                                url_story = new_story['url']
-                            else:
-                                url_story = '#'
-                            new_stories[new_story_id] = {
-                                'title': new_story['title'],
-                                'url': url_story
-                            }
-        _save_stories(new_stories)
+            new_stories = get_stories_of_topics()
+            if new_stories:
+                old_stories.update(new_stories)
+
+        _save_stories(old_stories)
```

Comment:
this code when you look at it, you think you are saving old_stories.

It is better to create new dictionary and return it, that makes code easier to read